sbi/neural_nets/mnle.py

```diff
@@ -30,28 +69,59 @@ def build_mnle(
 ):
     """Returns a density estimator for mixed data types.
 
-    Uses a categorical net to model the discrete part and a neural spline flow (NSF) to
-    model the continuous part of the data.
+    Uses a categorical net to model the discrete part and a conditional density
+    estimator to model the continuous part of the data.
+
+    Note: If the condition y is > 1D, an embedding net must be provided. Then,
+    during inference, we need to combine the embedded condition with the
+    discrete part of the input data. To do this, we use a combined embedding net
+    that takes the discrete part of the input and the embedded condition as
+    input and passes it on to the flow model.
+    To this end, we build the z-scoring and the embedding net in this function
+    here, i.e., outside of the flow model, and then pass z_score_x="none" to the
+    flow builder.
+    The y-embedding is passed to MixedDensityEstimator, which then uses it to
+    embed the continuous part and concatenate it with the discrete part during
+    log_prob evaluation and sampling.
+    The combined embedding net is passed to the flow model, which then uses it
+    to combine the already embedded and combined condition.
 
     Args:
-        batch_x: batch of data
-        batch_y: batch of parameters
-        z_score_x: whether to z-score x.
-        z_score_y: whether to z-score y.
-        num_transforms: number of transforms in the NSF
+        batch_x: Batch of xs, used to infer dimensionality.
+        batch_y: Batch of ys, used to infer dimensionality and (optional)
+            z-scoring.
+        z_score_x: Whether to z-score xs passing into the network, can be one of:
+            - `none`, or None: do not z-score.
+            - `independent`: z-score each dimension independently.
+            - `structured`: treat dimensions as related, therefore compute mean
+              and std
+            over the entire batch, instead of per-dimension. Should be used when
+            each sample is, for example, a time series or an image. For MNLE,
+            this applies to the continuous part of the data only.
+        z_score_y: Whether to z-score ys passing into the network, same options
+            as z_score_x.
+        flow_model: type of flow model to use for the continuous part of the
+            data.
+        embedding_net: Optional embedding network for y, required if y is > 1D.
+        combined_embedding_net: Optional embedding for combining the discrete
+            part of the input and the embedded condition into a joined
+            condition. If None, a linear embedding is used.
```

Comment:
I think we use a nonlinear embedding by default, maybe the docstring is wrong (or using the MLP is accidental below)?